Task4/Task4-1/main.cpp

```diff
@@ -0,0 +1,132 @@
+﻿// Task4-1.
+#include <iostream>
+#include <cmath>
+#include <string> 
+#include <random>   
+const int quantity = 81;
+using namespace std;
+/**
+*\ brief находит индексы тех элементов,значения которых больше значений последующего элемента
+*\ param  кол-во элементов массива,массив состоящий из целых чисел
+*\ return индексы тех элементов,значения которых больше значений последующего элемента
+*/
+string index(int *a, int quantity);
```

Comment:
не все параметры перечислены в документации
Почему `a` - ? -> `array`
Если параметры функции не изменяются, то они явно должны быть обозначены `const`
[By Alex Allain](https://www.cprogramming.com/tutorial/const_correctness.html)
>The const keyword allows you to specify whether or not a variable is modifiable. You can use const to prevent modifications to variables and const pointers and const references prevent changing the data pointed to (or referenced).
>Const дает вам возможность более четко документировать вашу программу и фактически обеспечивать соблюдение этой документации. Ключевое слово const обеспечивает вашим пользователям гарантии, позволяющие оптимизировать производительность без угрозы повреждения их данных, обеспечивая соблюдение вашей документации. Например, константные ссылки позволяют указать, что упомянутые данные не будут изменены; это означает, что вы можете использовать константные ссылки как простой и немедленный способ повышения производительности для любой функции, которая в настоящее время принимает объекты по значению, не беспокоясь о том, что ваша функция может изменить данные. Даже если это произойдет, компилятор предотвратит компиляцию кода и предупредит вас о проблеме. С другой стороны, если бы вы не использовали константные ссылки, у вас не было бы простого способа гарантировать, что ваши данные не были изменены.
Для индексации массивов и работы с их размерностью лучше использовать тип [`size_t`](https://en.cppreference.com/w/cpp/types/size_t). Для проверки корректности значения, которое будет использовано в качестве размера массива, нужно использовать тип [`int`](https://en.cppreference.com/w/cpp/language/types#Signed_and_unsigned_integer_types), например, следующим образом:

```cpp
size_t get_size(const std::string& message)
{
    int size = -1;
    std::cout << message;
    std::cin >> size;

    if (size < 0)
    {
        throw std::out_of_range("Incorrect size. Value has to be greater or equal zero.");
    }

    return size;
}
```